Catch Elastic's search_phase_execution and raise with descriptive message.

akkefa opened the pull request:
**Proposed changes**:
-  Catch Elastic's search_phase_execution_exception exception and raise with a descriptive message.


closes #1135

tholor reviewed (approved):
Thanks for your work @akkefa ! Just changed the error msg slightly and added a check as not all `RequestErrors` might be caused by missing embeddings. This check is also not 100% bullet proof but it should be the reason most of the times.

Ready to merge once CI has passed

akkefa commented:
> Thanks for your work @akkefa ! Just changed the error msg slightly and added a check as not all `RequestErrors` might be caused by missing embeddings. This check is also not 100% bullet proof but it should be the reason most of the times.
> 
> Ready to merge once CI has passed

Look good to me. Yes, I didn't notice the if statement. 🙈
